geode-core/src/jmh/java/org/apache/geode/internal/cache/tier/sockets/command/ClientCachePutBench.java

```diff
@@ -0,0 +1,233 @@
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Benchmark that measures throughput of single-threaded client performing puts to a loner server.
+ * <p>
+ * 100 random keys and values are generated during setup and the client reuses these in order,
+ * looping back around after 100 puts.
+ */
+@Measurement(iterations = 3, time = 2, timeUnit = MINUTES)
+@Warmup(iterations = 1, time = 1, timeUnit = MINUTES)
+@Fork(3)
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@State(Scope.Thread)
+@SuppressWarnings("unused")
+public class ClientCachePutBench {
+
+  static final String CLASS_NAME = ClientCachePutBench.class.getSimpleName();
+  static final String PACKAGE_NAME =
+      replace(ClientCachePutBench.class.getPackage().getName(), ".", "/");
+  static final String REGION_NAME = CLASS_NAME + "-region";
+  static final String SERVER_XML_NAME = "/" + PACKAGE_NAME + "/" + CLASS_NAME + "-server.xml";
+  static final long PROCESS_READER_TIMEOUT = 60 * 1000;
+  static final int NUMBER_OF_KEYS = 100;
+  static final int NUMBER_OF_VALUES = 100;
+
+  @State(Scope.Benchmark)
+  public static class ClientState {
+
+    Region<String, String> region;
+
+    String[] keys;
+    String[] values;
+
+    private int keyIndex = -1;
+    private int valueIndex = -1;
+
+    private Process process;
+    private volatile ProcessStreamReader processOutReader;
+    private volatile ProcessStreamReader processErrReader;
+
+    private int serverPort;
+    private ServerLauncher launcher;
+    private File serverDirectory;
+    private ClientCache clientCache;
+
+    private TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Setup(Level.Trial)
+    public void startServer() throws Exception {
+      System.out.println("\n" + "[ClientCachePutBench] startServer");
+
+      Random random = new Random(nanoTime());
+      this.keys = createStrings("KEY-", NUMBER_OF_KEYS, random);
+      this.values = createStrings("VAL-", NUMBER_OF_VALUES, random);
+
+      this.temporaryFolder.create();
+      this.serverDirectory = this.temporaryFolder.getRoot();
+
+      this.serverPort = getRandomAvailablePort(SOCKET);
+      this.process = startServerProcess(this.serverDirectory, this.serverPort);
+
+      try {
+        startProcessReaders(this.process);
+
+        ServerLauncher serverLauncher = new ServerLauncher.Builder()
+            .setWorkingDirectory(this.serverDirectory.getAbsolutePath()).build();
+
+        await("Starting server in " + this.serverDirectory).atMost(2, MINUTES)
+            .until(() -> assertThat(serverLauncher.status().getStatus()).isEqualTo(ONLINE));
+
+        this.clientCache = new ClientCacheFactory().set(LOG_LEVEL, "warn")
+            .addPoolServer(getIPLiteral(), this.serverPort).create();
+        this.region =
+            this.clientCache.<String, String>createClientRegionFactory(PROXY).create(REGION_NAME);
+
+      } finally {
+        stopProcessReaders(PROCESS_READER_TIMEOUT);
```

Comment:
I may be misreading this, but are we tearing down the process readers before we actually start the benchmark?